Mds/new mobile sdk/okta 503880 revise quickstart

mauricesharp-okta opened the pull request:
## Description:
Updated the iOS QuickStart for the updated Swift Mobile SDK (replaces OIDC).

> NOTE:
>
> The large number of files in the most recent commits are due to merging in changes from master.
>
> The **only** changes in this PR are to the files:
> - guides/sign-into-mobile-app-redirect/main/index.md
> - guides/sign-into-mobile-app-redirect/main/ios/*.md
> - vuepress-site/.vuepress/public/img/sign-users-in/redirect-authentication/ios/app-ui-ios.png

**Current Preview Build**
7/8 5:30PM Eastern (up to date with commits: rewrite getting user info)
https://62c89e3e4e21d6072335296a--reverent-murdock-829d24.netlify.app/docs/guides/sign-into-mobile-app-redirect/ios/main/#create-a-new-project

The only deltas in this preview are for [this commit](https://github.com/okta/okta-developer-docs/commit/0b19573d43326da867753a2353318cc07da7f17c)

**Changes**
The following are the changes to the included stack-selector files listed in the order that they appear in the main article.

`configmid.md`
- Updated prose and code
- Deleted the Singleton that checked configuration

`addconfigpkg.md`
- Updated the instructions for the new Swift package and package products

`createproject.md`
- Rewrote the section on getting user info to use the Token
- Updated prose and code
- Noted that the code uses `async`/`await` for asynchronous calls
- Moved all the code into the main view
- Added screenshots to show the UI for the signed-out and signed-in state: app-ui-ios.png
- Added a busy indicator that prevents tapping the UI by using `ZStack`
- Removed the `extension` that checked for a root view

`opensignin.md`
- Updated prose and code
- Updated sign in code with the simpler sign in call
- Added text to remind the developer to remove the credential from the store when the user signs out
- Added the `signOut` function (it was missing in the original version)

`getuserinfo.md`
- Updated prose and code
- Updated explanation for the new way of loading user info
- Changed the code to use the new `Credential` class and `userInfo` function.

`checkfortoken.md`
- Updated text and code for the new class and functions that make checking for an existing session easier

`refresh.md`
- Changed prose and code
- Added text for performance and battery life concerns in refreshing tokens
- Updated the code to use the much simpler method for refreshing a token

`usetoken.md`
- Updated prose and code for the much easier way to decorate a URL with the authentication token
- Documented alternative way to authorize the request using a synchronous call

### Resolves:

* [OKTA-503880](https://oktainc.atlassian.net/browse/OKTA-503880)

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/addconfigpkg.md:
```diff
@@ -2,5 +2,7 @@ Add the latest Okta OIDC package to your project:
-`https://github.com/okta/okta-oidc-ios.git`
-1. Confirm that the Dependency Rule is set to the main branch and that your project is selected, and then add the package.
+`https://github.com/okta/okta-mobile-swift.git`
+1. Confirm that the `okta-mobile-swift package` is selected, that the **Dependency Rule** is set to the main branch, and that your project is selected.
```
```suggestion
1. Confirm that the `okta-mobile-swift` package is selected, that the **Dependency Rule** is set to the main branch, and that your project is selected.
```

mauricesharp-okta replied:
The change is in the current commit

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
 Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.
 
-> **Note:** The sample code is compatible with iOS 14 and later.
+> **Note:** The sample code uses the async/await feature of iOS 15 and later. Replacing those with calls that use completion handlers adds compatibility for iOS 14 and later.
```
```suggestion
> **Note:** The sample code uses the async/await feature of iOS 15 and later. Replacing those with calls that use completion handlers adds compatibility for iOS 14 and earlier.
```

mauricesharp-okta replied:
Nice catch, it will be in the next commit.

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+The functionality of the app is implemented in the main SwiftUI view, including view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then using Combine, bindings, or `@Environment` for sharing state.
 
-Text views display the current state and any requested information, such as the access token. The available buttons depend on the sign-in state of the user. Update the imported modules and replace the `ContentView` `struct` with the following code:
+The text views in the app display its current state and any optional information, such as the access token. The available buttons depend on the sign-in state of the user. The top-level `ZStack` contains a full-screen view that prevents the user tapping the UI the app is performing certain asynchronous operations, such as reading the user information. A production app may also perform asynchronous ...
```
```suggestion
The text views in the app display its current state and any optional information, such as the access token. The available buttons depend on the sign-in state of the user. The top-level `ZStack` contains a full-screen view that prevents the user interacting with the UI while the app is performing certain asynchronous operations, such as reading the user information. A production app may also perform asynchronous operations that do require showing a busy indicator but don't require blocking the UI.
```

mauricesharp-okta replied:
Nice catch. It will be in the next commit

mikenachbaur-okta reviewed (approved):
I left some comments for suggestions, but other than that it all looks great!

JayNewstrom commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
-      .padding(.bottom)
-      ScrollView(showsIndicators: true) {
-        Text(infoText ?? "")
+      .alert(isPresented: $showingError) {
```
Is this a typo? It looks like `$showingError` wouldn't compile.

JayNewstrom commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -2,27 +2,28 @@ Display the user info by updating the `showUserInfo` function of `ContentView`:
+         busy = false
+         var userInfoText = ""
+         userInfo.payload.forEach { (key: String, value: Any) in
+            userInfoText += ("\(key): \(value) \n")
```
```suggestion
            userInfoText += ("\(key): \(value)\n")
```
Removing extra space.

brianduffield-okta reviewed (commented):
Nutrition Facts comment: 

- The number of learning outcomes is excessive and too granular, imo. I think they could be tightened up to 3 bullet points, which is standard across the docs
-  Missing a "What you need" section, which could contain, at least, your Okta org; it's referenced immediately in the Set up Okta section after the nutrition facts.

Create an Okta integration for your app comment:
- The intro blurb is very nice. As a team, we should standardize on this very common procedure.
- You provide both the CLI and Admin Console procedures here; I think you need to highlight that these are the same procedure: one from the command line and one from Admin Console (and that you only need to do this once).
- We've had some discussion as a group -- and probably need to have some more -- about the use of the classic "To" stem, which you've used here: "To create your app integration in Okta using the CLI". Although I've written thousands of these over the years, I think they're redundant and just adds unnecessary verbiage. I'd simply remove this and other like lines in the doc.

Passing off remainder of review to Brent

brentschaus-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
-Check for an existing session in two steps.
+To check for an existing session may take two steps:
```
If these steps are required, are they both required? If not, it's probably better to use bullet points.

mauricesharp-okta replied:
Nice catch! The numbered steps were from the older version which required both steps. I've updated them to bullets. That will appear on the next checkin.

brentschaus-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
-              signOut()
-            }
-            .padding(.bottom)
+   // Used for displaying error messages.
```
Refresh my memory re: comments in code - if it's a fragment, do we add a period?
```suggestion
   // Used for displaying error messages.
```

mauricesharp-okta replied:
I seem to remember from a previous article reviewed by Susan that the answer is "yes" and I can't find it in any of the various and sundry Okta style guides. I'm going to leave this for now. If there's a style issue, Susan will catch it in that final pass.

brentschaus-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -2,27 +2,28 @@ Display the user info by updating the `showUserInfo` function of `ContentView`:
 
-The completion handler in `getUser` is called when the Okta org server returns a result. Your production app may present a busy cursor and status while it waits for the callback. The completion handler then shows an error if appropriate or sorts the user info keys before displaying the raw data.
\ No newline at end of file
+The function first shows the activity indicator to block the UI, and then requests the credentials for the current user. The SDK calls the server and returns the user info array which is displayed in the info area.
```
```suggestion
The function first shows the activity indicator to block the UI, and then requests the credentials for the current user. The SDK calls the server and returns the user info array, which is displayed in the info area.
```

mauricesharp-okta replied:
Nice catch. Updated and will appear in next commit.

brentschaus-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md:
````diff
@@ -1,26 +1,44 @@
+}
+```
+
+The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
````
Is it worth mentioning here what the SDK displays if the sign-in is unsuccessful?

brentschaus-okta replied:
Or linking to relevant error handling content?

mauricesharp-okta replied:
For the first part, no as that's behavior that could change and has no impact on the code written by the developer.

I did think about looking for places to link to regarding error handling, but as a QuickStart I wanted to keep things to the minimum needed.

The other things is, link to what? If you do still think linking is needed, then what Okta content could I link to? In addition, on iOS the error code is either specific to Okta (defined in the SDK, in which case it's documented somewhere in the SDK so not linkable) or a generic iOS error such as a timeout, which is in the Apple documentation, (though specific error codes may be in different frameworks.) A more general "how to handle errors on iOS" also seems like something a developer should have access to on the Apple site... and external links are always a danger as things can change.

brentschaus-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/refresh.md:
```diff
@@ -1,39 +1,60 @@
+
+Most of the code is for handling errors or controling the state of the busy view.
+
+Tokens are usually refreshed on a regular basis. Each refresh requires network activity which is one of the operations that requires higher battery use. In your production app the best practice is to refresh a token only when it's required to authenticate a server call. Some other considerations for refreshing a token include:
```
```suggestion
Tokens are usually refreshed on a regular basis. Each refresh requires network activity, which is one of the operations that requires higher battery use. In your production app, the best practice is to refresh a token only when it's required to authenticate a server call. Some other considerations for refreshing a token include:
```

mauricesharp-okta replied:
Can you tell this was one of the latter ones I revised ;-). Thanks for catching those. The changes will be in the next commit.

brentschaus-okta replied:
I guess it happens to the best of us.

brentschaus-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/refresh.md:
```diff
@@ -1,39 +1,60 @@
+- Checking if the device is using Data rather than WiFi.
+- Handling failed refresh attempts, especially when the server can't be reached.
+
+The `Token` object contains information such as expiry date. The following function displays the access token, it's issue and expiry dates, and any associated refresh token in the info area:
```
```suggestion
The `Token` object contains information such as expiry date. The following function displays the access token, its issue and expiry dates, and any associated refresh token in the info area:
```

mauricesharp-okta replied:
Another nice one and another change in the next commit.

brentschaus-okta reviewed (changes requested):
Looks great, Maurice.

I have a few, small nits and a few questions. Please feel free to reach out to me if you need to.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
 #### Add the UI
 
-The main view implements all of the app functionality, including initializing the Okta SDK. Note that best practice for a full app uses a singleton to manage the interaction with the SDK.
+The functionality of the app is implemented in the main SwiftUI view, including view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then using Combine, bindings, or `@Environment` for sharing state.
```
Much nicer wording. I did modify it a little as I realized I forgot to refer to the file containing the view, and to tighten it up a bit, how about...

```suggestion
The functionality of the app is implemented in `ContentView.swift`, the main SwiftUI view. This includes view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then sharing state using Combine, bindings, or `@Environment`.
```

brentschaus-okta replied:
Works for me.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+Source image(s): https://www.figma.com/file/i3huE0gEoISu2evquOq5yJ/app-ui-redirect-ios?node-id=0%3A1
+There's only one board. The group for the image is: "Side-by-side screenshots"
+-->
+
```
Yep, I know about the lack of space. I asked Chris and he said he'd add a margin to the styles... i.e., don't do anything about it myself either with a `<br/>` or `div style`.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+There's only one board. The group for the image is: "Side-by-side screenshots"
+-->
+
+Open `ContentView.swift` and replace it with the following code:
```
Nice catch. I slightly modified it as I realized the answer is "neither". It's only replacing any import statements and the ContentView struct. Replacing the contents of the file would also remove the preview struct as well. How's this:

```suggestion
Open `ContentView.swift` and use the following code to replace the `import` statements and the `ContentView` struct:```

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/addconfigpkg.md:
```diff
@@ -2,5 +2,7 @@ Add the latest Okta OIDC package to your project:
+`https://github.com/okta/okta-mobile-swift.git`
+1. Confirm that the `okta-mobile-swift` package is selected, that the **Dependency Rule** is set to the main branch, and that your project is selected.
+1. Click **Add Package**.
+1. Select all of the package products in the dialog that appears: **AuthFoundation**, **OktaOAuth2**, and **WebAuthenticationUI**, and click **Add Package** to add the package and products to your project.
```
Another nice catch. Change made for next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
 
-1. Initialize the authorization state manager (`authStateManager`) from a saved session.
-2. Verify that the access token is still valid by checking the expiry.
+1. Check for an existing valid token.
```
Change made based on earlier comment, though I used "-" instead of "*" for the bullet... paranoia since "*" is also used for emphasis and bold.

brentschaus-okta replied:
True :)

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
-1. Initialize the authorization state manager (`authStateManager`) from a saved session.
-2. Verify that the access token is still valid by checking the expiry.
+1. Check for an existing valid token.
+1. Refresh the token if it has expired.
```
Same comment as above.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
+1. Refresh the token if it has expired.
 
-The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` which only tries to refresh the token ...
```
I'll take "that" ;-). Change made and will be in the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
 Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.
 
-> **Note:** The sample code is compatible with iOS 14 and later.
+> **Note:** The sample code uses features of SwiftUI available in iOS 14, and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and later.
```
In this case I didn't use code font because that's the way Apple presents the term on their developer website. At the moment I haven't made this change.

brentschaus-okta replied:
Makes good sense. Thanks!

mauricesharp-okta commented:
@brianduffield-okta. FYI, none of that material was changed in this update and was previously approved. The original choice for the list in outcomes was that this is the QuickStart, so we wanted to show the developer all the things they'd see. I don't mind reducing the content, do you have any suggestion as to what it should be?

For the CLI vs. Admin console, it's not providing two flows and there were no changes in this PR. It's providing the CLI flow and adding information on how to see what's been done in the console. The intent of the paragraph between the CLI and next piece is to show that: "At this point, you can move to the next step, Creating your app..." If that's not clear, then do you have any suggestions. 

And for the "To..." stem I completely agree. It's there because it wasn't there when I revised this article and I was told to add it. Unless you have an objection, I'll leave that for Susan's pass.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -10,37 +12,3 @@ Create the Okta configuration file and add the keys and values for your applicat
    | `logoutRedirectUri` | The post-logout redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/` |
```
Easier to give it the name in the Admin Console?

```suggestion
   | `logoutRedirectUri` | The sign-out redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/` |
```

mauricesharp-okta replied:
Done. In the next commit.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -10,37 +12,3 @@ Create the Okta configuration file and add the keys and values for your applicat
    | `logoutRedirectUri` | The post-logout redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/` |
    | `redirectUri` | The Redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/callback` |
```
Easier to give it the name in the Admin Console?

```suggestion
   | `redirectUri` | The Sign-in redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/callback` |
```

mauricesharp-okta replied:
Done. In the next commit.

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md:
````diff
@@ -1,26 +1,44 @@
+}
+```
+
+The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
````
```suggestion
The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in flow is successful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

mauricesharp-okta replied:
in the next commit

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md:
````diff
@@ -1,26 +1,44 @@
+}
+```
+
+The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
````
sign-in shouldn't be used as a noun acc2 the style guide.

```suggestion
The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in attempt is successful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

mauricesharp-okta replied:
Done. In the next commit.

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md:
```diff
@@ -1,26 +1,44 @@
+The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in is succesful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
 
-  authSession?.signInWithBrowser(from: rootViewController) { authStateManager, error in
+The stored token is used to check if the user is already signed in when the app is launched, and for API calls to the Okta org and authorization servers.
```
```suggestion
The stored token is used to check if the user is already signed in when the app is launched and for API calls to the Okta org and authorization servers.
```

mauricesharp-okta replied:
Thanks. In the next commit.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -2,27 +2,28 @@ Display the user info by updating the `showUserInfo` function of `ContentView`:
 
```
I can't comment the top line for some reason. For consistency, should you have `ContentView.swift` rather than just `ContentView` or should it say "the `ContentView` struct"?

mauricesharp-okta replied:
Good catch. Changed to the file name. In the next commit.

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
+               busy = false
+            }
+         } else {
+            // There is existing session, hide the activity indicator.
```
```suggestion
            // If there is an existing session, hide the activity indicator.
```

mauricesharp-okta replied:
A great catch, though it may be different text is required. The semantics of this else clause are that there is no existing session. Perhaps a comment of:
// There is no existing session, hide the activity indicator

susanharper-okta replied:
makes perfect sense to me, thanks.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
+- Refresh the token if it has expired.
 
-The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token  ...
```
Why have you started adding `()` to the function names in the text here? There are many references to `signIn`, `signOut`, `showError` etc earlier on that don't have the signature part.

mauricesharp-okta replied:
This may be partially a revision issue and also one to do with a mix of code the developer writes versus code in the SDK. I'm not sure which direction to go on this and would love to hear what you think. In the above text I use the full signature of the SDK function because it's from the SDK and there are other "store" functions with different signatures.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
+- Refresh the token if it has expired.
 
-The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the users' credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token  ...
```
```suggestion
The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
```

mauricesharp-okta replied:
Nice catch "There can be only one." –Highlander. Change is in the next commit.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/refresh.md:
````diff
@@ -1,39 +1,60 @@
+}
+```
+
+Most of the code is for handling errors or controling the state of the busy view.
````
```suggestion
Most of the code is for handling errors or controlling the state of the busy view.
```

mauricesharp-okta replied:
I hat it whon I meighk spalling errers. In the next commit.

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+      showingError = true
+   }
+
+   // A utility function to update the state of all visual elements.
```
```suggestion
   // A utility function that updates the state of all visual elements.
```

mauricesharp-okta replied:
Thanks. In the next commit.

susanharper-okta reviewed (commented):
couple things. thank you!

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/refresh.md:
```diff
@@ -1,39 +1,60 @@
+- Checking if the device is using Data rather than WiFi.
+- Handling failed refresh attempts, especially when the server can't be reached.
+
+The `Token` object contains information such as expiry date. The following function displays the access token, its issue and expiry dates, and any associated refresh token in the info area:
```
Can't see why this is capitalised...
Also, what is the info area? You mention it only once before and don't explain it.

```suggestion
The `token` object contains information such as expiry date. The following function displays the access token, its issue and expiry dates, and any associated refresh token in the info area:
```

mauricesharp-okta replied:
because it should have said "class", not object. That change is in the next commit.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -2,27 +2,28 @@ Display the user info by updating the `showUserInfo` function of `ContentView`:
 
-The completion handler in `getUser` is called when the Okta org server returns a result. Your production app may present a busy cursor and status while it waits for the callback. The completion handler then shows an error if appropriate or sorts the user info keys before displaying the raw data.
\ No newline at end of file
+The function first shows the activity indicator to block the UI, and then requests the credentials for the current user. The SDK calls the server and returns the user info array, which is displayed in the info area.
```
What is the info area? This is the first time you've mentioned it.

mauricesharp-okta replied:
Fixed in previous comment

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
-Check for an existing session in two steps.
+To check for an existing session may take two steps:
```
```suggestion
When you check for an existing session, it may take two steps:
```

mauricesharp-okta replied:
Thanks. In the next commit.

danielmaharry-okta reviewed (changes requested):
Hi @mauricesharp-okta, a dozen or so minor nits to fix. Some may be me not understanding Swift \ iOS development correctly. There are a handful of corrections to index.md though that I couldn't comment idrectly on because you hand't changed it.

1. index.md Line 106 - 107. In the admin console, these are called `Sign-in redirect URI` and `Sign-out redirect URI` (and definitely not "logout" anything :-)
2. index.md Line 164 I don't understand what you mean by "Your own Server API". Is this some API the developer must have separately? Is it the IDX API? Could you clarify please? 
3. testapp.md Line 1. Does Simulator need to be capitalised?
4. testapp.md line 4. "after" is spelled "Afer"
5. More generally, you seem to add function signatures consistently from "Check for a session at startup" but they are mostly missing up to that point.  
6. Last but not least, I flagged this up in the text, but you mention "the info area" a couple of times but never explain what you mean by the term.

Other than that, it reads well so far.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -1,3 +1,5 @@
+The iOS SDK uses information from the application integration you created earlier to connect to the Okta org. The SDK loads these values from a configuration file named `Okta.plist` if it exists in your main bundle. You may also specify the values in code using `WebAuthentication(issuer:clientId:scopes:responseType:redirectUri:logoutRedirectUri:additionalParameters:)`.
```
Nice catch. It will be in the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
 Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.
 
-> **Note:** The sample code is compatible with iOS 14 and later.
+> **Note:** The sample code uses features of SwiftUI available in iOS 14, and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and later.
```
Thanks! It's in the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+The functionality of the app is implemented in `ContentView.swift`, the main SwiftUI view. This includes view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then sharing state using Combine, bindings, or `@Environment`.
 
-Text views display the current state and any requested information, such as the access token. The available buttons depend on the sign-in state of the user. Update the imported modules and replace the `ContentView` `struct` with the following code:
+The text views in the app display its current state and any optional information, such as the access token. The available buttons depend on the sign-in state of the user. The top-level `ZStack` contains a full-screen view that prevents the user interacting with the UI while the app is performing certain asynchronous operations, such as reading the user information. A production app may also perfo ...
```
Done. In the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
-              signOut()
-            }
-            .padding(.bottom)
+   // Used for displaying error messages.
```
Thanks. In the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+   @State var errorMessage: String? = nil
+   @State var errorTitle: String? = nil
+
+   // Text shown in the status area.
```
Thanks. In the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md:
```diff
@@ -1,26 +1,44 @@
-      error: error)
-      return
-    }
+Remove the stored credential when the user signs-out. Replace the `signOut` function with the following code:
```
Thanks. In the next commit.

mauricesharp-okta commented:
Hi @ danielmaharry-okta,

For these items:
> 1. index.md Line 106 - 107. In the admin console, these are called `Sign-in redirect URI` and `Sign-out redirect URI` (and definitely not "logout" anything :-)
Done

> 2. index.md Line 164 I don't understand what you mean by "Your own Server API". Is this some API the developer must have separately? Is it the IDX API? Could you clarify please?
This was written by Chris. I've no idea what it means either.

> 3. testapp.md Line 1. Does Simulator need to be capitalised?
Depends on whether this refers to Simulator the application or a simulator in Simulator ;-). I'm choosing the former and changing this to:

1. Build and run your app in Simulator or on a device.


> 4. testapp.md line 4. "after" is spelled "Afer"
Nice catch, fixed in the next commit.

> 5. More generally, you seem to add function signatures consistently from "Check for a session at startup" but they are mostly missing up to that point.
Can you say more? Give some examples?

> 6. Last but not least, I flagged this up in the text, but you mention "the info area" a couple of times but never explain what you mean by the term.
I modified the paragraph starting at line 9 of `createproject.md`. If necessary I could add callouts to the screenshots, though would prefer to avoid that if at all possible as the images are already approved:

"The app contains three areas: the top area contains buttons for signing in and out, and for viewing information, the middle area shows the current app state, and the lower area displays information text, such as the token. The available buttons depend on the sign-in... "

> Other than that, it reads well so far.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
 #### Add the UI
 
-The main view implements all of the app functionality, including initializing the Okta SDK. Note that best practice for a full app uses a singleton to manage the interaction with the SDK.
+The functionality of the app is implemented in `ContentView.swift`, the main SwiftUI view. This includes view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then sharing state using Combine, bindings, or `@Environment`.
```
It's an Apple trademark that refers to a well known technology. I chose not to link to Apple developer documentation because it's both off site, and the possibility of links changing.

So, yes, it requires a capital and the expectation is that iOS developers will know what it means. Unless you have an objection, I'm leaving it as is.

danielmaharry-okta replied:
OK then that's jsut me not knowing Apple tech. This is fine.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,144 @@
+There's only one board. The group for the image is: "Side-by-side screenshots"
+-->
+
+Open `ContentView.swift` and use the following code to replace the `import` statements and the `ContentView` struct:
```
It's more than the struct, it's also the import statements. This is important because a SwiftUI file has at least 3 parts that should be there: the framework import statements, the struct defining the view, the preview struct. With that last item there's no dynamic preview in the development environment. I didn't want to add those lines of code as they're not relevant to the task at hand.

That's why it's replacing the import statements and view struct. If you'd still like to change the wording, do you have a suggestion? Or is OK as is?

danielmaharry-okta replied:
I don't think switching the text around as suggested makes it any less obvious that both are being replaced because otherwise you appear to be omitting what to repalce the import statements with. I don't see anything about using an ampersand in the style guide, so 

> replace the `import` statements & the `ContentView` struct with the following code:

joins the two together more obviously. If that still doesn't work for you, leave as is.

mauricesharp-okta replied:
OK... this is what's in the next commit:

> Open `ContentView.swift` and replace the `import` statements and the `ContentView` struct with the following code:

danielmaharry-okta replied:
That's lovely. Thank you.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
````diff
@@ -1,154 +1,144 @@
 ```
 
-The last four functions are utilities. You fill out `tokenExpired` later. The two variants of `showError` configure the contents, and then show an alert (the `showingError` state variable controls the presentation of the alert.)
+The last three functions are utilities. The two variants of `showError` set the state variables for the title and text of an alert, then set `showingError` which triggers the `.alert` view modifier to present the alert.
````
Change is in next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/addconfigpkg.md:
```diff
@@ -2,5 +2,7 @@ Add the latest Okta OIDC package to your project:
+`https://github.com/okta/okta-mobile-swift.git`
+1. Confirm that the `okta-mobile-swift` package is selected, that the **Dependency Rule** is set to the main branch, and that your project is selected.
+1. Click **Add Package**.
+1. Select all of the package products in the dialog that appears: **AuthFoundation**, **OktaOAuth2**, and **WebAuthenticationUI**, and then click **Add Package** to add the package and products to your project.
```
Nice catch... and I'm usually the one cutting words ;-). Change is in the next commit.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -1,3 +1,5 @@
+The iOS SDK uses information from the application integration you created earlier to connect to the Okta org. The SDK loads these values from a configuration file named `Okta.plist` if it exists in your main bundle. You may also specify the values in code using `WebAuthentication(issuer:clientId:scopes:responseType:redirectUri:logoutRedirectUri:additionalParameters:)`.
```
The dangers of stack selectors... especially when you didn't write the main content. I removed the first sentence and and modified the second:
"The SDK loads the configuration values from..."

So far I'm planning on putting that in the next commit. Let me know if it should be different.

danielmaharry-okta replied:
No thats fine

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -1,3 +1,5 @@
+The iOS SDK uses information from the application integration you created earlier to connect to the Okta org. The SDK loads these values from a configuration file named `Okta.plist` if it exists in your main bundle. You may also specify the values in code using `WebAuthentication(issuer:clientId:scopes:responseType:redirectUri:logoutRedirectUri:additionalParameters:)`.
+
 Create the Okta configuration file and add the keys and values for your application integration:
```
Not really. by default it will be created in the active directory, which is the top level of the project contents. However, all plist files are moved into the default bundle during compilation unless they are specifically assigned to a different bundle.

danielmaharry-okta replied:
ok - then never mind.

mauricesharp-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/refresh.md:
```diff
@@ -1,39 +1,60 @@
+You refreshed a token when checking for an existing session in [Check for a session at startup](#check-for-a-session-at-startup). In this case you may want to catch any errors that occur.
+
 Update `refreshToken` in `ContentView` with the following code:
```
I added the file extension, we should make a choice on the function signatures.

danielmaharry-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -9,6 +9,6 @@ Create the Okta configuration file and add the keys and values for your applicat
-   | `logoutRedirectUri` | The post-logout redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/` |
-   | `redirectUri` | The Redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/callback` |
+   | `logoutRedirectUri` | The sign-out redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/` |
+   | `redirectUri` | The Sign-in redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/callback` |
```
Sorry but you had sign-out in line above and Sign-in here. Couldn't take it.....

```suggestion
   | `redirectUri` | The sign-in redirect URI from the app integration that you created, such as `com.okta.dev-1234567:/callback` |
```

danielmaharry-okta commented:
Hi @mauricesharp-okta I've approved your changes. Just found two more things that are tweaks. Line 13 of configmid.md should be "sign-in", not "Sign-in" and redirectvalues.md was hiding two more values that should read "Sign-in redirect URI" and "Sign-out redirect URI".

> More generally, you seem to add function signatures consistently from "Check for a session at startup" but they are mostly missing up to that point. Can you say more? Give some examples?

createproject.md line 142, opensignin.md line 1, 24, getuserinfo.md line 1, refresh.md line 3

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/index.md:
```diff
@@ -126,9 +126,11 @@ The SDK signs in the user by opening an Okta-hosted web page. The app may send t
+The ID token returned by Okta contains user information, or *claims*, that are based on the scopes requested by the app (see [Configure your app](#configure-your-app)).
 
-The default profile items (called `claims`) returned by Okta include the user's email address, name, and preferred username. The claims that you see may differ depending on what scopes your app has requested (see [Configure your app](#configure-your-app)).
+This app includes the `profile` scope which includes the user's email address, name, and preferred username. You can use this information to update your UI, such as showing the customer's name.
```
```suggestion
This app includes the `profile` scope that includes the user's email address, name, and preferred username. You can use this information to update your UI, such as showing the customer's name.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/index.md:
```diff
@@ -126,9 +126,11 @@ The SDK signs in the user by opening an Okta-hosted web page. The app may send t
-The default profile items (called `claims`) returned by Okta include the user's email address, name, and preferred username. The claims that you see may differ depending on what scopes your app has requested (see [Configure your app](#configure-your-app)).
+This app includes the `profile` scope which includes the user's email address, name, and preferred username. You can use this information to update your UI, such as showing the customer's name.
+
+Use the the Okta user information endpoints for items that aren't available in the ID token. For general information on requesting user info, see the [userinfo response example](/docs/reference/api/oidc/#response-example-success-6)).
```
```suggestion
Use the Okta user information endpoints for items that aren't available in the ID token. For general information on requesting user info, see the [userinfo response example](/docs/reference/api/oidc/#response-example-success-6)).
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
-1. Initialize the authorization state manager (`authStateManager`) from a saved session.
-2. Verify that the access token is still valid by checking the expiry.
+- Check for an existing valid token.
+- Refresh the token if it has expired.
```
```suggestion
- Refresh the token if it's expired.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/checkfortoken.md:
```diff
@@ -1,36 +1,42 @@
+- Refresh the token if it has expired.
 
-The state manager session and its associated token are already saved by the `signIn` function calling `writeToSecureStorage`.
+The `Credential` class manages the tokens for a user. The `signIn()` function you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential has expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token  ...
```
```suggestion
The `Credential` class manages the tokens for a user. The `signIn()` function that you wrote in [Open the sign-in page](#open-the-sign-in-page) calls `store(_:)` to save the user's credentials. A session exists if there's a default credential. Although there are calls to check if a credential is expired and to request a refresh, this code uses `refreshIfNeeded()` that only tries to refresh the token if it's expired.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,149 @@
 Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.
 
-> **Note:** The sample code is compatible with iOS 14 and later.
+> **Note:** The sample code uses SwiftUI features that are available in iOS 14, and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and later.
```
```suggestion
> **Note:** The sample code uses SwiftUI features that are available in iOS 14 and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and later.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,149 @@
+The functionality of the app is implemented in `ContentView.swift`, the main SwiftUI view. This includes view state, view model updates, and data model. For a production app, consider partitioning the view and data models into singletons, and then sharing state using Combine, bindings, or `@Environment`.
 
-Text views display the current state and any requested information, such as the access token. The available buttons depend on the sign-in state of the user. Update the imported modules and replace the `ContentView` `struct` with the following code:
+The app contains three areas: the top area contains buttons for signing in and out, and for viewing information, the middle area shows the current app state, and the lower area displays information text, such as the token. The available buttons depend on the sign-in state of the user. The top-level `ZStack` contains a full-screen view that prevents the user from interacting with the UI while the  ...
```
```suggestion
The app contains three areas: the top area contains buttons for signing in and out and for viewing information. The middle area shows the current app state, and the lower area displays information text, such as the token. The available buttons depend on the sign-in state of the user. The top-level `ZStack` contains a full-screen view that prevents the user from interacting with the UI while the app is performing certain asynchronous operations, such as reading the user information. A production app may also perform asynchronous operations that do require showing a busy indicator but don't require blocking the UI.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
````diff
@@ -1,154 +1,149 @@
 ```
 
-The last four functions are utilities. You fill out `tokenExpired` later. The two variants of `showError` configure the contents, and then show an alert (the `showingError` state variable controls the presentation of the alert.)
+The last three functions are utilities. The two overloads of `showError` set the state variables for the title and text of an alert, then set `showingError` which triggers the `.alert` view modifier to present the alert.
````
```suggestion
The last three functions are utilities. The two overloads of `showError` set the state variables for the title and text of an alert, then set `showingError` that triggers the `.alert` view modifier to present the alert.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -1,28 +1,25 @@
 
-The completion handler in `getUser` is called when the Okta org server returns a result. Your production app may present a busy cursor and status while it waits for the callback. The completion handler then shows an error if appropriate or sorts the user info keys before displaying the raw data.
\ No newline at end of file
+The function reads the user information from the local ID token. The available information depend on the scopes that are specified in the `Okta.plist` file. You can also request information from the server with the `userInfo()` of the `Credential` class.
```
```suggestion
The function reads the user information from the local ID token. The available information depends on the scopes that are specified in the `Okta.plist` file. You can also request information from the server with the `userInfo()` of the `Credential` class.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/redirectvalues.md:
```diff
@@ -1,2 +1,2 @@
-* **Redirect URI**: Use the default
-* **Post Logout Redirect URI(s)**: Use the default
\ No newline at end of file
+* **Sign In Redirect URI**: Use the default
```
```suggestion
* **Sign In Redirect URI**: Use the default.
```

susanharper-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/redirectvalues.md:
```diff
@@ -1,2 +1,2 @@
-* **Post Logout Redirect URI(s)**: Use the default
\ No newline at end of file
+* **Sign In Redirect URI**: Use the default
+* **Sign Out Redirect URI(s)**: Use the default
```
```suggestion
* **Sign Out Redirect URI(s)**: Use the default.
```

susanharper-okta reviewed (approved):
couple nits approving to not hold you up. Awful lot of other files in the PR.

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/addconfigpkg.md:
```diff
@@ -2,5 +2,7 @@ Add the latest Okta OIDC package to your project:
-`https://github.com/okta/okta-oidc-ios.git`
-1. Confirm that the Dependency Rule is set to the main branch and that your project is selected, and then add the package.
+`https://github.com/okta/okta-mobile-swift.git`
+1. Confirm that the `okta-mobile-swift` package is selected, that the **Dependency Rule** is set to the main branch, and that your project is selected.
```
We currently are stuck using the branch named `master`.  Does that matter to this?

mauricesharp-okta replied:
I'll explain why I did this and if you still think it needs changing, let me know.

I chose "main" because it reflects inclusive language. I deliberately didn't use code voice because the meaning then becomes "the main branch of the repository". I think developers understand that for most repositories that's still "master", but it also means that when we can finally change it, no documentation change is required.

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/configmid.md:
```diff
@@ -1,3 +1,5 @@
+The SDK loads the configuration values from a configuration file named `Okta.plist` if it exists in your main bundle. You may also specify the values in code using `WebAuthentication(issuer:clientId:scopes:responseType:redirectUri:logoutRedirectUri:additionalParameters:)`.
```
```suggestion
The SDK loads the configuration values from a configuration file named `Okta.plist` if it exists in your main bundle. You may also specify the values in code using `WebAuthentication(issuer:clientId:scopes:redirectUri:logoutRedirectUri:additionalParameters:)`.
```

Recent updates dropped the `responseType:` argument.

mauricesharp-okta replied:
Updated in next commit

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/createproject.md:
```diff
@@ -1,154 +1,149 @@
 Launch Xcode and create a new iOS app project using SwiftUI for the interface and Swift for the language.
 
-> **Note:** The sample code is compatible with iOS 14 and later.
+> **Note:** The sample code uses SwiftUI features that are available in iOS 14 and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and later.
```
This makes it sound like only iOS 14+ is supported, when in fact older versions are supported.

```suggestion
> **Note:** The sample code uses SwiftUI features that are available in iOS 14 and the async/await feature of iOS 15. Use the completion handler version of the async/await functions to add compatibility with iOS 14 and earlier.
```

mauricesharp-okta replied:
I think we had this same exchange before. Although the SDK works with earlier versions, this sample won't. It uses features of SwiftUI that are iOS 14 or later, which is why the initial part of the sentence is there. If that's not clear then I need to revise the second sentence... how about:

>  Use the completion handler version of the async/await functions to enable the sample to run on iOS 14 and later.

mikenachbaur-okta replied:
I like that better, it makes it clearer that it's talking about the sample app itself.

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -1,28 +1,25 @@
-  authStateManager.getUser() { response, error in
-    if let error = error {
-      showError(title: "Unable to Read User Info", error: error)
+   guard let credential = Credential.default,
+         let idToken = credential.token.idToken else {
```
This can be simplified, since the `credential` variable isn't being used elsewhere in this function.

```suggestion
   guard let idToken = Credential.default?.token.idToken else {
```

mauricesharp-okta replied:
Nice catch... the dangers of updating functions for new ways of doing things and not paying attention to the whole thing... I used to use `credential` when it was a server call.  The change will be in the next commit.

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/getuserinfo.md:
```diff
@@ -1,28 +1,25 @@
 
-The completion handler in `getUser` is called when the Okta org server returns a result. Your production app may present a busy cursor and status while it waits for the callback. The completion handler then shows an error if appropriate or sorts the user info keys before displaying the raw data.
\ No newline at end of file
+The function reads the user information from the local ID token. The available information depends on the scopes that are specified in the `Okta.plist` file. You can also request information from the server with the `userInfo()` of the `Credential` class.
```
It might be nice to clarify what information is being requested with the userInfo call.

```suggestion
The function reads the user information from the local ID token. The available information depends on the scopes that are specified in the `Okta.plist` file. You can also request the full user's profile from the server with the `userInfo()` of the `Credential` class.
```

mauricesharp-okta replied:
AFAIK the issue is that it returns only the claims that are consistent with the scopes. "full user profile" isn't actually descriptive because it can mean many things. The only way to give a somewhat accurate description of what's returned is to talk about the underlying endpoints called by `userInfo()`, but that assumes those won't change.

Thoughts?

mikenachbaur-okta replied:
Hmm, how about simply "You can also request additional information about the user from the server with the ....."

mauricesharp-okta replied:
How about something slightly shorter:

> You can also request user information from the server...

mikenachbaur-okta commented on packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/ios/opensignin.md:
````diff
@@ -1,26 +1,44 @@
+}
+```
+
+The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in flow is successful, the app first stores the token using `Credential.store(_)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
````
```suggestion
The app hands the sign-in flow to the Okta SDK. The SDK presents the sign-in web view and waits for a result. If the sign-in flow is successful, the app first stores the token using `Credential.store(_:)`, and then updates the user interface by setting the `signedIn` state variable to `true`.
```

mauricesharp-okta replied:
Nice catch. It's in the next commit.

mauricesharp-okta replied:
I'm adding that wording to a commit, so please let me know if it's OK
